students/aaron/lesson02/assignment/src/charges_calc.py

```diff
@@ -0,0 +1,133 @@
+
+    return parser.parse_args()
+
+
+def load_rentals_file(filename):
+    """
+    Loads a json file
+    """
+    LOGGER.debug("load_rentals_file()")
+    try:
+        with open(filename) as file:
+            LOGGER.debug("Loading JSON data from %s", filename)
+            data = json.load(file)
+            LOGGER.debug("Successfully loaded JSON data.")
+    except Exception as err:
+        LOGGER.error("Failed to load JSON data. %s", err)
+        raise err
+    return data
+
+def calculate_additional_fields(data):
+    """
+    Calculates additional fields
+    """
+    keys = ['rental_start', 'rental_end', 'price_per_day', 'units_rented']
+    for value in data.values():
+        missing_keys = [x for x in keys if not x in value.keys()]
+        if missing_keys:
+            LOGGER.warning("Could not find required key(s) %s in %s.  Skip.",
+                           missing_keys,
+                           value)
+            continue
+        try:
+            rental_start = datetime.datetime.strptime(value['rental_start'],
+                                                      '%m/%d/%y')
+            rental_end = datetime.datetime.strptime(value['rental_end'],
+                                                    '%m/%d/%y')
+        except ValueError as err:
+            LOGGER.warning("Failed to calculate time from %s or %s. %s",
+                           value['rental_start'],
+                           value['rental_end'],
+                           err)
+        if value['units_rented'] == 0:
+            LOGGER.error("Units rented is 0 for %s.  Skip.", value)
+            continue
+        if rental_start >= rental_end:
+            LOGGER.warning("Rental start %s on or after end %s in %s.  Skip.",
+                           rental_start,
+                           rental_end,
+                           value)
+            continue
+        try:
+            value['total_days'] = (rental_end - rental_start).days
+            value['total_price'] = value['total_days'] * value['price_per_day']
+            value['sqrt_total_price'] = math.sqrt(value['total_price'])
+            value['unit_cost'] = value['total_price'] / value['units_rented']
+        except Exception as err:
+            LOGGER.error("Failed to calculate fields for %s. %s", value, err)
+            raise err
+
+    return data
+
+def save_to_json(filename, data):
+    """
+    Saves data in JSON format to a file
+    """
+    try:
+        with open(filename, 'w') as file:
+            json.dump(data, file)
+    except Exception as err:
+        LOGGER.error("Failed to write output to %s. %s", filename, err)
+        raise err
+
+LOG_MAP = {'1': logging.ERROR,   # errors
+           '2': logging.WARNING, # errors and warnings
+           '3': logging.DEBUG}   # errors, warnings, and debug
+
+if __name__ == "__main__":
+    ARGS = parse_cmd_arguments()
+    if ARGS.debug and ARGS.debug in LOG_MAP:
+        LOGGER = setup_logger(LOG_MAP[ARGS.debug])
```

Comment:
OK

Reply:
I use the LOG_MAP to safely handle command line input for the --debug flag.  It works well for this situation because there's a very limited number of valid inputs (0, 1, 2, or 3) .  If the input is anything other than the allowed keys, it defaults to the same behavior as if they had passed 0.